sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2574,6 +2574,17 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val BUCKETING_COALESCE_ENABLED =
+    buildConf("spark.sql.bucketing.coalesce")
```

Comment:
How about `spark.sql.bucketing.coalesce`->`....coalesceBucketInJoin`, `...coalesceJoinedBucket`, ...  based on the naming rule? https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/internal/config/ConfigEntry.scala#L41-L42

Reply:
Changed to `coalesceBucketsInJoin`